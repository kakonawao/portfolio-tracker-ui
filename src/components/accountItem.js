import React from "react";


class AccountItem extends React.Component {
    render() {
        return (
            <div>
                <p>{this.props.code}: {this.props.description} ({this.props.holder ? this.props.holder.name : "N/A"})</p>
            </div>
        )
    }
}

export default AccountItem;
