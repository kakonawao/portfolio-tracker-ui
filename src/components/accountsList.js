import React from "react";

import AccountItem from "./accountItem";


class AccountsList extends React.Component {

    renderLoading() {
        return (
            <p>Loading accounts...</p>
        )
    }

    renderEmpty() {
        return (
            <p>You have no accounts.</p>
        )
    }

    renderList(items) {
        return (
            items.map(i =>
                <AccountItem code={i.code} description={i.description} holder={i.holder}/>
            )
        );
    }

    render () {
        if (this.props.data.loading) {
            return this.renderLoading();
        } else if (this.props.data.items.length === 0) {
            return this.renderEmpty();
        } else {
            return this.renderList(this.props.data.items);
        }
    }
}

export default AccountsList;
