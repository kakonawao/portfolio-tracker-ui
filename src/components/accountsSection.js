import React from "react";
import { connect } from "react-redux";

import { fetchAccounts } from "../actions/accounts";
import AccountsList from "./accountsList";


const mapStateToProps = (state) => {
    return {
        session: state.session,
        data: state.accounts
    }
};

export class AccountsSection extends React.Component {

    constructor(props) {
        super(props);

        this.props.dispatch(fetchAccounts(this.props.session));
    }

    render() {
        return (
            <div>
                <h2>Accounts</h2>
                <AccountsList data={this.props.data}/>
            </div>
        )
    }
}

export default connect(mapStateToProps)(AccountsSection);
