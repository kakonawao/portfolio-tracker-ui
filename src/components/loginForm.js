import React from "react";


class LoginForm extends React.Component {

    render() {
        return (
            <form>
                <div>
                    <label>
                        Username
                        <input type={"text"}
                               name={"username"}
                               ref={this.props.username} />
                    </label>
                    <label>
                        Password
                        <input type={"password"}
                               name={"password"}
                               ref={this.props.password} />
                    </label>
                </div>
                <div>
                    <button type={"submit"}
                            onClick={this.props.login}>
                        Login
                    </button>
                </div>
            </form>
        );
    }
}

export default LoginForm;
