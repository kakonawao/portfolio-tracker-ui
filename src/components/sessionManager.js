import React from "react";
import { connect } from "react-redux";

import LoginForm from "./loginForm";
import { login } from "../actions/session";


const mapStateToProps = (state) => {
    return {
        authenticated: state.session.authenticated,
        username: state.session.username
    }
};

class SessionManager extends React.Component {

    constructor(props) {
        super(props);

        this.inputs = {
            username: React.createRef(),
            password: React.createRef()
        }
    }

    login = (event) => {
        event.preventDefault();
        const loginData = {
            username: this.inputs.username.current.value,
            password: this.inputs.password.current.value
        };
        this.props.dispatch(login(loginData));
    };

    render() {
        return (
            <div>
                <h2>Login</h2>
                <p>
                    You need to log in to use this service.
                </p>
                <LoginForm login={this.login}
                           username={this.inputs.username}
                           password={this.inputs.password}/>
            </div>
        );
    };
}

export default connect(mapStateToProps)(SessionManager);
