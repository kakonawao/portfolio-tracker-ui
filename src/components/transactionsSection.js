import React from "react";


export class TransactionsSection extends React.Component {

    render() {
        return (
            <div>
                <h2>Transactions</h2>
            </div>
        );
    }
}

export default TransactionsSection;
