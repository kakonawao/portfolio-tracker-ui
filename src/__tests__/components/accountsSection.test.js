import React from "react";
import { shallow } from "enzyme";

import { AccountsSection } from "../../components/accountsSection";

describe("<AccountsSection>", () => {

    it("renders account list sub component", () => {
        const props = {
            session: {
                loading: false,
                authenticated: false,
                username: null,
                token: null,
                tokenType: null
            },
            data: {
                loading: true,
                items: []
            },
            dispatch: jest.fn()
        };

        const wrapper = shallow(<AccountsSection {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });
});
