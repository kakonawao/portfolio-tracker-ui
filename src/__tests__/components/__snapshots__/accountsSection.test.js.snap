// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountsSection> renders account list sub component 1`] = `
<div>
  <h2>
    Accounts
  </h2>
  <AccountsList
    data={
      Object {
        "items": Array [],
        "loading": true,
      }
    }
  />
</div>
`;
